frontend/app/(tabs)/home.tsx

```diff
@@ -59,17 +107,58 @@ const Index = () => {
     loadInitialCards();
   }, []);
 
+  // Animation styles
+  const searchBarScale = searchAnimation.interpolate({
+    inputRange: [0, 1],
+    outputRange: [1, 1.015],
+  });
+
   return (
     <View className="flex-1 bg-primary">
+      {/* Simple SearchBar */}
+      <View className="px-6 pt-6 pb-3">
+        <Animated.View
+          style={{
+            transform: [{ scale: searchBarScale }],
+          }}
+          className="flex-row items-center rounded-2xl overflow-hidden border border-gray-100 bg-white/90 shadow-lg"
+        >
+          <TextInput
+            className="flex-1 py-4 px-6 text-lg font-medium text-gray-800"
+            placeholder="Search idioms..."
+            value={searchText}
+            onChangeText={handleSearch}
+            onFocus={handleFocus}
+            placeholderTextColor="#9CA3AF"
+          />
+          {searchText !== '' && (
+            <TouchableOpacity onPress={clearSearch} className="pr-5">
+              <Text className="text-gray-400 text-xl">✕</Text>
```

Comment:
Is this the best way to do this? I would have imagined that you use an icon instead of a text with an x.

Reply:
Maybe we can consider using a component library that already implements such a control